Source/cmMakefile.cxx

```diff
@@ -181,18 +181,9 @@ void cmMakefile::PrintCommandTrace(const cmListFileFunction& lff) const
     this->GetCMakeInstance()->GetTraceFiles();
   std::string const& full_path = this->GetExecutionFilePath();
   std::string const& only_filename = cmSystemTools::GetFilenameName(full_path);
-  bool trace = true;
-  if (!trace_only_this_files.empty()) {
-    trace = false;
-    for (std::vector<std::string>::const_iterator i =
-        trace_only_this_files.begin();
-        i != trace_only_this_files.end(); ++i) {
-      if (only_filename == *i) {
-        trace = true;
-        break;
-      }
-    }
-  }
+  bool trace = trace_only_this_files.empty() ||
+    std::find(trace_only_this_files.begin(), trace_only_this_files.end(),
+      only_filename) != trace_only_this_files.end();
```

Comment:
FYI, CMake does not use C++11 (without support being hidden behind `#ifdef`'d macros) because it still supports being built compilers which do not have any C++11 support (e.g., Visual Studio 9 2008).